pyquil/tests/test_resource_manager.py

```diff
@@ -0,0 +1,90 @@
+from pyquil.resource_manager import *
+import pyquil.quil as pq
+from pyquil.gates import *
+import pytest
+import json
+from mock import Mock
+import numpy as np
+
+
+@pytest.fixture
+def five_qubit_prog():
+    p = pq.Program()
+    qubits = [p.alloc() for _ in xrange(5)]
+    p.inst(map(H, qubits))
+    return p, qubits
+
+
+def test_alloc():
+    p, qubits = five_qubit_prog()
+    for qubit in qubits:
+        assert qubit in p.resource_manager.live_qubits
+        assert not instantiated(qubit)
+        check_live_qubit(qubit)
+    # Give the qubits labels
+    for qubit in qubits:
+        p.resource_manager.instantiate(qubit)
+        assert instantiated(qubit)
+    for qubit in qubits:
+        p.free(qubit)
+        assert qubit in p.resource_manager.dead_qubits
+
+
+def test_add_qubits():
+    p = pq.Program()
+    q = pq.Program()
+    p_qubits = [p.alloc(), p.alloc()]
+    q_qubits = [q.alloc(), q.alloc()]
+    p.inst(H(p_qubits[0]), CNOT(*p_qubits))
+    q.inst(H(q_qubits[0]), CNOT(*q_qubits))
+    two_bell = p + q
+    assert p.resource_manager.live_qubits + q.resource_manager.live_qubits == \
```

Comment:
This is pedantic but I don't think we should rely on order here. But it's probably fine to get away with it here.